client/src/controllers/UnsavedController.ts

```diff
@@ -66,19 +66,24 @@ const DEFAULT_DURATIONS = {
  */
 export class UnsavedController extends Controller<HTMLFormElement> {
   static values = {
-    confirmation: { default: '', type: String },
+    confirmation: { default: false, type: Boolean },
     durations: { default: DEFAULT_DURATIONS, type: Object },
     force: { default: false, type: Boolean },
     hasComments: { default: false, type: Boolean },
     hasEdits: { default: false, type: Boolean },
     watch: { default: 'edits', type: String },
   };
 
-  /** Translated value for the beforeunload confirmation dialog, if empty no confirmation will show. */
-  declare confirmationValue: string;
+  /** Whether to show the browser confirmation dialog. */
+  declare confirmationValue: boolean;
```

Comment:
Browsers have no longer displayed a custom message since a long long time ago:

https://caniuse.com/mdn-api_window_beforeunload_event_generic_string_displayed

So we shouldn't use a string for this.

Reply:
When doing the original refactor we did not want to change this behaviour as I deemed it not related to the Stimulus PR.

I agree though we should align with the modern standard now that we do not need to support older browsers.